File: packages/java/parser-jvm-plugin-transfertypes/src/test/java/com/vaadin/hilla/parser/plugins/transfertypes/pageable/basic/PageableEndpoint.java
package com.vaadin.hilla.parser.plugins.transfertypes.pageable.basic;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Slice;
import org.springframework.data.domain.Sort;

@Endpoint
public class PageableEndpoint {
    public Sort.Order getOrder() {
        return null;
    }

    public Page<String> getPage() {
        return null;
    }

    public Slice<String> getSlice() {
        return null;
    }

    public Pageable getPageable() {
        return null;
    }

    public Sort getSort() {
        return null;
    }

    public Custom getCustom() {
        return null;
    }

    public static class Custom {
        public Page page;
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/transfertypes/pageable/basic/snapshots/PageableEndpoint.ts
import type { EndpointRequestInit } from '@vaadin/hilla-frontend';
import type Order from './com/vaadin/hilla/mappedtypes/Order.js';
import type Pageable from './com/vaadin/hilla/mappedtypes/Pageable.js';
import type Sort from './com/vaadin/hilla/mappedtypes/Sort.js';
import client from './connect-client.default.js';

export async function getOrder(init?: EndpointRequestInit): Promise<Order | undefined> {
  return client.call('PageableEndpoint', 'getOrder', {}, init);
}

export async function getPage(init?: EndpointRequestInit): Promise<Array<string | undefined> | undefined> {
  return client.call('PageableEndpoint', 'getPage', {}, init);
}

export async function getPageable(init?: EndpointRequestInit): Promise<Pageable | undefined> {
  return client.call('PageableEndpoint', 'getPageable', {}, init);
}

export async function getSort(init?: EndpointRequestInit): Promise<Sort | undefined> {
  return client.call('PageableEndpoint', 'getSort', {}, init);
}
